### PR TITLE
main/iptables: upgrade to 1.8.2

### PR DESCRIPTION
**Pull request by J0WI**

### CODE REVIEW

File: main/iptables/iptables-1.8.2-musl-fixes.patch [hunk 1]
```diff
@@ -34,12 +34,27 @@
...
  #include <linux/netfilter_ipv4.h>
+
+--- a/extensions/libebt_vlan.c
++++ b/extensions/libebt_vlan.c
+@@ -13,7 +13,11 @@
+ #include <ctype.h>
+ #include <xtables.h>
+ #include <linux/netfilter_bridge/ebt_vlan.h>
++#if defined (__GLIBC__)
+ #include <linux/if_ether.h>
```
Comment (nmeum) [inline main/iptables/iptables-1.8.2-musl-fixes.patch]: `linux/if_ether.h` should be a header from `linux-headers` and not related to GLIBC at all. Which makes me wonder why this patch is necessary?

Comment (J0WI) [reply main/iptables/iptables-1.8.2-musl-fixes.patch]: Upstream patch is here: https://git.netfilter.org/iptables/commit/?id=51d374ba41ae4f1bb851228c06b030b83dd2092f
Last time there was also a conflict with ebtables: https://github.com/alpinelinux/aports/commit/ad7465682e465adc4d142ffabb09d51b0e159f53

**Comment by ncopa** [comment]
iirc recent iptables broke things at runtime and had to be reverted. commit ad7465682e465adc4d142ffabb09d51b0e159f53 says that it conflicts with ebtables. Has this been resolved?

**Comment by J0WI** [comment]
iptables aims to replace ebtables:
https://git.netfilter.org/iptables/commit/?id=632ace7c2947dbb70f74ea263b86ff68de391622
https://git.netfilter.org/iptables/commit/?id=f988fe49839015c6e798a1d4e805aaf52691dbf7
Should I add `conflicts="ebtables"`?

**Comment by ncopa** [comment]
`libvirt` currently depends on both `ebtables` and `iptables` so with this pr `libvirt` will break. We need to clean that up at the same time. Maybe it is as easy as remove `ebtables` package, add those lines to `iptables`:
````sh
provides="ebtables" # for backards compat
replaces="ebtables"
````
and modify the `libvirt` package:
````diff
diff --git a/main/libvirt/APKBUILD b/main/libvirt/APKBUILD
index 985c43e5c0..42c77e718c 100644
--- a/main/libvirt/APKBUILD
+++ b/main/libvirt/APKBUILD
@@ -7,7 +7,7 @@ pkgdesc="A virtualization API for several hypervisor and container systems"
 url="http://libvirt.org/"
 arch="all"
 license="LGPL"
-_daemon_deps="bridge-utils dmidecode dnsmasq ebtables ip6tables iptables"
+_daemon_deps="bridge-utils dmidecode dnsmasq ip6tables iptables"
 _client_deps="pm-utils gnutls-utils netcat-openbsd"
 depends="lvm2"
 depends_dev="libtirpc-dev"
````
And then verify that `libvirt` is not obviously broken.

But we cannot merge this as is.

**Comment by J0WI** [comment]
Closed in 5639562d54068646cdd2ba00beb75688043a1210

**Pull request closed without merging**